### ISSUE TITLE
Idea: To distinguish easier between needuml and needarch, needuml shall not be stored inside a need

### ISSUE THREAD

**Issue by PhilipPartsch** [useblocks/sphinx-needs#749]
Currently: needuml is been stored inside a need, like needarch. 
Idea: We provide this only with needarch.

**Comment by danwos** [comment]
I totally agree and I thought this is already the case.
At least it was the goal.

``needuml``: Pure representation, no way to reuse it somewhere else
``needarch``: Storing and representation, can be reused by ``needuml`` and ``needarch``.

**Comment by PhilipPartsch** [comment]
Todos:
1. exception in needuml if a key is given
    good place for the exception is in [here ](https://github.com/useblocks/sphinx-needs/blob/a7b725e2fce34e2adf6bd7238e48897fa8e23b25/sphinx_needs/directives/needuml.py#L76)
3. test that needuml is raising the new defined exception
4. test that needarch still works
5. check in copy function form needuml to need, only needarch are been copied.
[Implementation of copy from needuml to need](https://github.com/useblocks/sphinx-needs/blob/029387ef8448c8ef6c551edd6759d024ccbe8bab/sphinx_needs/api/need.py#L459)
6. Adapt documentation for needuml and needarch

**Comment by AlexanderLanin** [comment]
We have two keywords for the same functionality which are used in two different places.... it really sounds like there should be only one such keyword! Focus on the end user, not how it's easier internally. For the end user it's one less keyword they need to learn.

**Comment by danwos** [comment]
I disagree here, as they behave slightly differently and have access to different kind of data.
The decision was made because of the user and not based on any technical implementation, as it is hard to explain why the same code behaves differently, has different jinja-functions support, and does not have access to the same data, based on where it was used.  

In the past, it was one single function, but users starts to complain about it quite fast. Therefore we decided to split it, so that also the documentation can be done more easily.
